src/main/kotlin/racinggame/application/RacingService.kt

```diff
@@ -13,22 +13,24 @@ object RacingService {
         names: List<String>,
         moveCondition: MoveCondition,
         roundCount: Int,
-    ) {
-        require(!names.isEmpty()) { ERROR_CAR_COUNT }
+    ): RaceResult {
+        require(names.isNotEmpty()) { ERROR_CAR_COUNT }
         require(roundCount > 0) { ERROR_ROUND_COUNT }
 
-        val cars = MutableList<Car>(names.size) { i -> Car(names[i], 0) }
+//        val cars = MutableList(names.size) { i -> Car(names[i], 0) }
 
+        val cars =
+            ArrayDeque<Car>().apply {
+                names.forEach { name ->
+                    add(Car(name, 0))
+                }
+            }
```

Comment:
```suggestion
            val cars = ArrayDeque(names.map(::Car))
```

이렇게도 표현 가능할 것 같네요. 모두 공통으로 0이라는 숫자가 들어간다면 내부적으로 처리해도 될 것 같구요. 
forEach는 간단한 출력이나 디버깅 용도로 간단한 list.add작업이 안되는건아니지만 좀 더 적절한 로직으로 map같은 오퍼레이터가 있으니 이를 활용해보면 좋을 것 같네요